darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -216,6 +223,337 @@ def test_manual_save_and_load(self):
+                model_new_enc_past_n_future.load_weights(
+                    model_path_manual, load_encoders=False, map_location="cpu"
+                )
+
+        def test_save_and_load_weights_w_likelihood(self):
+            """
+            Verify that save/load does not break likelihood.
+
+            Note: since load_weights() calls load_weights_from_checkpoint(), it will be used
+            for all but one test.
+            Note: Using DLinear since it supports both past and future covariates
+            """
+
+            def create_DLinearModel(
+                model_name: str,
+                save_checkpoints: bool = False,
+                likelihood: Likelihood = None,
+            ):
+                return DLinearModel(
+                    input_chunk_length=4,
+                    output_chunk_length=1,
+                    kernel_size=5,
+                    model_name=model_name,
+                    work_dir=self.temp_work_dir,
+                    save_checkpoints=save_checkpoints,
+                    likelihood=likelihood,
+                    random_state=42,
+                    force_reset=True,
+                )
+
+            model_dir = os.path.join(self.temp_work_dir)
+            manual_name = "save_manual"
+            auto_name = "save_auto"
+            # create manually saved model checkpoints folder
+            checkpoint_path_manual = os.path.join(model_dir, manual_name)
+            os.mkdir(checkpoint_path_manual)
+            checkpoint_file_name = "checkpoint_0.pth.tar"
+            model_path_manual = os.path.join(
+                checkpoint_path_manual, checkpoint_file_name
+            )
+
+            model_auto_save = create_DLinearModel(
+                auto_name,
+                save_checkpoints=True,
+                likelihood=GaussianLikelihood(prior_mu=0.5),
+            )
+            model_auto_save.fit(self.series, epochs=1)
+            pred_auto = model_auto_save.predict(n=4, series=self.series)
+
+            model_manual_save = create_DLinearModel(
+                manual_name,
+                save_checkpoints=False,
+                likelihood=GaussianLikelihood(prior_mu=0.5),
+            )
+            model_manual_save.fit(self.series, epochs=1)
+            model_manual_save.save(model_path_manual)
+            pred_manual = model_manual_save.predict(n=4, series=self.series)
+
+            # predictions are identical when using the same likelihood
+            self.assertTrue(np.array_equal(pred_auto.values(), pred_manual.values()))
+
+            # model with identical likelihood
+            model_same_likelihood = create_DLinearModel(
+                "same_likelihood", likelihood=GaussianLikelihood(prior_mu=0.5)
+            )
+            model_same_likelihood.load_weights(model_path_manual, map_location="cpu")
+            model_same_likelihood.to_cpu()
+            model_same_likelihood.predict(n=4, series=self.series)
```

Comment:
We can check that `load_weights_from_checkpoint` and `load_weights` have equal predictions.

Reply:
The Likelihood object do not have the same `random_state` (also mentioned in #1779), making the predictions different. Should I implement a method to set it before updating this test?

Reply:
see my suggestion below